frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -373,49 +473,67 @@ function MainMenu(props: Props): ReactElement {
     }
   }, [] as any[])
 
-  const shouldShowS4AMenu = !!S4AMenuOptions.length
+  const shouldShowS4AMenu = !!S4AMenuItems.length
   const showDeploy = isLocalhost() && !shouldShowS4AMenu && props.canDeploy
   const showSnapshot = !shouldShowS4AMenu && props.sharingEnabled
-  const showClearCache = !shouldShowS4AMenu
   const preferredMenuOrder: any[] = [
-    coreMenuOptions.rerun,
-    showClearCache && coreMenuOptions.clearCache,
-    shouldShowS4AMenu && coreMenuOptions.settings,
-    coreMenuOptions.DIVIDER,
-    showDeploy && coreMenuOptions.deployApp,
-    showSnapshot && coreMenuOptions.saveSnapshot,
-    coreMenuOptions.recordScreencast,
-    ...(shouldShowS4AMenu
-      ? S4AMenuOptions
-      : [
-          coreMenuOptions.DIVIDER,
-          coreMenuOptions.documentation,
-          coreMenuOptions.community,
-          coreMenuOptions.report,
-          coreMenuOptions.DIVIDER,
-          coreMenuOptions.s4t,
-          coreMenuOptions.settings,
-          coreMenuOptions.about,
-        ]),
+    coreMenuItems.rerun,
+    coreMenuItems.settings,
+    coreMenuItems.DIVIDER,
+    coreMenuItems.recordScreencast,
+    showSnapshot && coreMenuItems.saveSnapshot,
+    coreMenuItems.DIVIDER,
+    coreMenuItems.report,
+    coreMenuItems.community,
+    ...(shouldShowS4AMenu ? S4AMenuItems : [coreMenuItems.DIVIDER]),
+    coreMenuItems.about,
+  ]
+
+  const preferredDevMenuOrder: any[] = [
+    coreDevMenuItems.developerOptions,
+    coreDevMenuItems.clearCache,
+    showDeploy && coreDevMenuItems.deployApp,
+    isLocalhost() && coreDevMenuItems.s4t,
+    coreDevMenuItems.reportSt,
+    coreDevMenuItems.documentation,
+    coreDevMenuItems.visitStForum,
   ]
 
   // Remove empty entries, and add dividers into menu options as needed.
-  const menuOptions: any[] = []
+  const menuItems: any[] = []
   let lastMenuItem = null
   for (const menuItem of preferredMenuOrder) {
     if (menuItem) {
-      if (menuItem !== coreMenuOptions.DIVIDER) {
-        if (lastMenuItem === coreMenuOptions.DIVIDER) {
-          menuOptions.push({ ...menuItem, hasDividerAbove: true })
+      if (menuItem !== coreMenuItems.DIVIDER) {
+        if (lastMenuItem === coreMenuItems.DIVIDER) {
+          menuItems.push({ ...menuItem, hasDividerAbove: true })
         } else {
-          menuOptions.push(menuItem)
+          menuItems.push(menuItem)
         }
       }
 
       lastMenuItem = menuItem
     }
   }
 
+  const devMenuItems: any[] = []
+  let devLastMenuItem = null
+  for (const devMenuItem of preferredDevMenuOrder) {
+    if (devMenuItem) {
+      if (devMenuItem !== coreDevMenuItems.DIVIDER) {
+        if (devLastMenuItem === coreDevMenuItems.DIVIDER) {
+          devMenuItems.push({ ...devMenuItem, hasDividerAbove: true })
+        } else {
+          devMenuItems.push(devMenuItem)
+        }
+      }
+
+      devLastMenuItem = devMenuItem
+    }
+  }
```

Comment:
nothing actionable here since this wasn't originally your code, but this really needs a refactor 😱